opendcs-rest-api/src/main/java/org/opendcs/odcsapi/sec/AuthorizationCheck.java

```diff
@@ -15,23 +15,103 @@
 {
-	String ODCS_API_GUEST = "ODCS_API_GUEST";
-	String ODCS_API_USER = "ODCS_API_USER";
-	String ODCS_API_ADMIN = "ODCS_API_ADMIN";
+	public static final String ODCS_API_GUEST = "ODCS_API_GUEST";
+	public static final String ODCS_API_USER = "ODCS_API_USER";
+	public static final String ODCS_API_ADMIN = "ODCS_API_ADMIN";
+	private static final Logger LOGGER = LoggerFactory.getLogger(AuthorizationCheck.class);
 
 	/**
 	 * Authorizes the current session returning the SecurityContext that will check user roles.
 	 *
 	 * @param requestContext     context for the current session.
 	 * @param httpServletRequest context for the current request.
 	 */
-	SecurityContext authorize(ContainerRequestContext requestContext, HttpServletRequest httpServletRequest);
+	public abstract SecurityContext authorize(ContainerRequestContext requestContext,
+			HttpServletRequest httpServletRequest, ServletContext servletContext);
+
+	public abstract boolean supports(String type, ContainerRequestContext requestContext);
+
+
+	protected final ApiAuthorizationDAI getAuthDao(ServletContext servletContext)
+	{
+		OpenDcsDatabase db = createDb(servletContext);
+		TimeSeriesDb timeSeriesDb = db.getLegacyDatabase(TimeSeriesDb.class)
+				.orElseThrow(() -> new UnsupportedOperationException("Endpoint is unsupported by the OpenDCS REST API."));
+		//Need to figure out a better way to extend the toolkit API to be able to add dao's within the REST API
+		if(timeSeriesDb instanceof CwmsTimeSeriesDb)
+		{
+			return new CwmsAuthorizationDAO(timeSeriesDb);
+		}
+		else if(timeSeriesDb instanceof OpenTsdb)
+		{
+			return new OpenTsdbAuthorizationDAO(timeSeriesDb);
+		}
+		throw new UnsupportedOperationException("Endpoint is unsupported by the OpenDCS REST API.");
+	}
 
-	boolean supports(String type, ContainerRequestContext requestContext);
+	private OpenDcsDatabase createDb(ServletContext servletContext)
+	{
+		DataSource dataSource = (DataSource) servletContext.getAttribute(DATA_SOURCE_ATTRIBUTE_KEY);
+		try
+		{
+			if(dataSource == null)
+			{
+				throw new IllegalStateException("No data source defined in context.xml");
+			}
+			return DatabaseService.getDatabaseFor(dataSource);
+		}
+		catch(DatabaseException e)
+		{
+			//Temporary workaround until database_properties table is implemented in the schema
+			LOGGER.atWarn().setCause(e).log("Temporary solution forcing OpenTSDB");
+			DecodesSettings decodesSettings = new DecodesSettings();
+			decodesSettings.CwmsOfficeId = System.getProperty("DB_OFFICE");
+			try(Connection connection = dataSource.getConnection())
+			{
+				DatabaseProvider databaseProvider;
+				String databaseProductName = connection.getMetaData().getDatabaseProductName();
+				if(databaseProductName.toLowerCase().startsWith("oracle"))
+				{
+					databaseProvider = new CwmsDatabaseProvider();
+				}
+				else
+				{
+					databaseProvider = new OpenTsdbProvider();
+				}
+				return databaseProvider.createDatabase(dataSource, decodesSettings);
```

Comment:
any particular reason the DatabaseService factory doesn't work? i suspect the decodessettings + office issue but want to confirm.

Reply:
turned into "how much code do I want to write that will get thrown away anyways"

Reply:
Would it make sense to allow passing a DecodesSettings, or even just a `java.util.Properties` in even with the DataSource.

Like, eventually with a merge which the passed in one would override anything in the database?

Reply:
That's what I was considering but couldn't think of the use-case outside of this temporary solution. Merging properties from the file system, process runtime, etc with the database seems like asking for a maintenance headache.

Reply:
Yeah. I can kinda see a case for it, but how often would it come up? Though the merge I had in mind was a simple (if just using java.util.Properties) passed in value of same name wins. But for now we'll leave it. Not like it's super difficult to add such support later if we decide it's required.